### ISSUE TITLE
[ilcorsaronero] Found no results while trying to browse this tracker (Test)

### ISSUE THREAD

**Issue by injNinja** [Jackett/Jackett#1970]
The log isn't telling me anything that can be useful, i just dont get anything from this tracker, from one day to another.

I can reach ilcorsaronero.info from my browser, ping and telnet.

Help!

**Comment by garfield69** [comment]
Have you turned on the enhanced logging? Set it on the Jackett Dashboard
![image](https://user-images.githubusercontent.com/6933240/31598048-0dd784ca-b2a8-11e7-914d-b21e655d12f0.png)
repeat the test and see if the logs have something useful.
More info on where to find the complete log files (as opposed to the dashboard log view) https://github.com/Jackett/Jackett#troubleshooting
(edit out personal details (username, password, apikey) and then 
drag and drop the log here so we can take a look

**Comment by injNinja** [comment]
yesterday it didn’t show me anything, tomorrow i figured out it was just a certificate validation error.
added the chain manually into the docker container and all went fine! thank you anyway!
a-

Inviato da iPhone

**Comment by bubez81** [comment]
sorry, i don't have the docker, how can i solve?

**Comment by kaso17** [comment]
@bubez81 please provide a full log

**Comment by bubez81** [comment]
ok:
`

Oct 17, 2017 01:28:45 PM | Info | ReflectedHttpActionDescriptor ExecuteAsync Invoking action 'Logs()'
-- | -- | --
Oct 17, 2017 01:28:45 PM | Info | ApiControllerActionInvoker InvokeActionAsync Action='Logs()'
Oct 17, 2017 01:28:45 PM | Info | JackettAPINoCacheAttribute OnActionExecutingAsync
Oct 17, 2017 01:28:45 PM | Info | JackettAPINoCacheAttribute OnActionExecutingAsync Action filter for 'Logs()'
Oct 17, 2017 01:28:45 PM | Info | HttpActionBinding ExecuteBindingAsync
Oct 17, 2017 01:28:45 PM | Info | HttpActionBinding ExecuteBindingAsync
Oct 17, 2017 01:28:45 PM | Info | JackettAuthorizedAttribute OnAuthorizationAsync
Oct 17, 2017 01:28:45 PM | Info | JackettAuthorizedAttribute OnAuthorizationAsync
Oct 17, 2017 01:28:45 PM | Info | ApiControllerActionSelector SelectAction Selected action 'Logs()'
Oct 17, 2017 01:28:45 PM | Info | ApiControllerActionSelector SelectAction
Oct 17, 2017 01:28:45 PM | Info | ServerConfigurationController ExecuteAsync
Oct 17, 2017 01:28:45 PM | Info | HttpControllerDescriptor CreateController Jackett.Controllers.V20.ServerConfigurationController
Oct 17, 2017 01:28:45 PM | Info | DefaultHttpControllerActivator Create Jackett.Controllers.V20.ServerConfigurationController
Oct 17, 2017 01:28:45 PM | Info | DefaultHttpControllerActivator Create
Oct 17, 2017 01:28:45 PM | Info | HttpControllerDescriptor CreateController
Oct 17, 2017 01:28:45 PM | Info | DefaultHttpControllerSelector SelectController ServerConfiguration
Oct 17, 2017 01:28:45 PM | Info | DefaultHttpControllerSelector SelectController Route='action:logs,controller:ServerConfiguration'
Oct 17, 2017 01:28:45 PM | Info | http://bubez81.dyndns.org:9117/api/v2.0/server/logs?_=1508239410028
Oct 17, 2017 01:28:38 PM | Info | IndexerApiController Dispose
Oct 17, 2017 01:28:38 PM | Info | IndexerApiController Dispose
Oct 17, 2017 01:28:38 PM | Info | Content-type='application/json; charset=utf-8', content-length=35322
Oct 17, 2017 01:28:38 PM | Info | IndexerApiController ExecuteAsync
Oct 17, 2017 01:28:38 PM | Info | ApiExceptionHandler ExecuteExceptionFilterAsync
Oct 17, 2017 01:28:38 PM | Error | {0}
Oct 17, 2017 01:28:38 PM | Info | ApiExceptionHandler ExecuteExceptionFilterAsync
Oct 17, 2017 01:28:38 PM | Info | JackettAPINoCacheAttribute OnActionExecutedAsync
Oct 17, 2017 01:28:38 PM | Info | JackettAPINoCacheAttribute OnActionExecutedAsync Action filter for 'Test()'
Oct 17, 2017 01:28:38 PM | Info | RequiresIndexerAttribute OnActionExecutedAsync
Oct 17, 2017 01:28:38 PM | Info | RequiresIndexerAttribute OnActionExecutedAsync Action filter for 'Test()'
Oct 17, 2017 01:28:38 PM | Error | ApiControllerActionInvoker InvokeActionAsync
Oct 17, 2017 01:28:38 PM | Error | ReflectedHttpActionDescriptor ExecuteAsync
Oct 17, 2017 01:28:38 PM | Debug | IWebClient.GetString(Url:https://ilcorsaronero.info/recenti)
Oct 17, 2017 01:28:38 PM | Info | ReflectedHttpActionDescriptor ExecuteAsync Invoking action 'Test()'
Oct 17, 2017 01:28:38 PM | Info | ApiControllerActionInvoker InvokeActionAsync Action='Test()'
Oct 17, 2017 01:28:38 PM | Info | RequiresIndexerAttribute OnActionExecutingAsync
Oct 17, 2017 01:28:38 PM | Info | RequiresIndexerAttribute OnActionExecutingAsync Action filter for 'Test()'
Oct 17, 2017 01:28:38 PM | Info | JackettAPINoCacheAttribute OnActionExecutingAsync
Oct 17, 2017 01:28:38 PM | Info | JackettAPINoCacheAttribute OnActionExecutingAsync Action filter for 'Test()'
Oct 17, 2017 01:28:38 PM | Info | HttpActionBinding ExecuteBindingAsync
Oct 17, 2017 01:28:38 PM | Info | HttpActionBinding ExecuteBindingAsync
Oct 17, 2017 01:28:38 PM | Info | JackettAuthorizedAttribute OnAuthorizationAsync
Oct 17, 2017 01:28:38 PM | Info | JackettAuthorizedAttribute OnAuthorizationAsync
Oct 17, 2017 01:28:38 PM | Info | ApiControllerActionSelector SelectAction Selected action 'Test()'
Oct 17, 2017 01:28:38 PM | Info | ApiControllerActionSelector SelectAction
Oct 17, 2017 01:28:38 PM | Info | IndexerApiController ExecuteAsync
Oct 17, 2017 01:28:38 PM | Info | HttpControllerDescriptor CreateController Jackett.Controllers.V20.IndexerApiController
Oct 17, 2017 01:28:38 PM | Info | DefaultHttpControllerActivator Create Jackett.Controllers.V20.IndexerApiController
Oct 17, 2017 01:28:38 PM | Info | DefaultHttpControllerActivator Create
Oct 17, 2017 01:28:38 PM | Info | HttpControllerDescriptor CreateController
Oct 17, 2017 01:28:38 PM | Info | DefaultHttpControllerSelector SelectController IndexerApi

`

**Comment by kaso17** [comment]
can you please post the log file instead of the online log view output?

**Comment by bubez81** [comment]
sure sorry
[log.txt](https://github.com/Jackett/Jackett/files/1391049/log.txt)

**Comment by kaso17** [comment]
```2017-10-17 14:56:09.5576 Error Jackett.IndexerException: Exception (ilcorsaronero): An error occurred while sending the request ---> System.Net.Http.HttpRequestException: An error occurred while sending the request ---> System.Net.WebException: Error: TrustFailure (Ssl error:1000007d:SSL routines:OPENSSL_internal:CERTIFICATE_VERIFY_FAILED) ---> Mono.Btls.MonoBtlsException: Ssl error:1000007d:SSL routines:OPENSSL_internal:CERTIFICATE_VERIFY_FAILED```
This is often caused by missing CA certificates.
Try reimporting the certificates:
   - On Linux (as user root): `wget -O - https://curl.haxx.se/ca/cacert.pem | cert-sync /dev/stdin`
   - On macOS: `curl -sS https://curl.haxx.se/ca/cacert.pem | cert-sync --user /dev/stdin`

**Comment by kaso17** [comment]
Sorry, this is a dupe of #1431

**Comment by kaso17** [comment]
With the latest jackett version it should work

**Comment by bubez81** [comment]
strange, doesn't work 
` encoding: UTF-8
  links:
    - https://ilcorsaronero.info/
  certificates:
    - FFB230323B987B07C24DCC13BC892CF30536F015 # incomplete CA chain

`
nether 
curl -sS https://curl.haxx.se/ca/cacert.pem | cert-sync --user /dev/stdin

**Comment by maxliscia** [comment]
Same issue here, if it helps this website uses Let's Encrypt ssl certificate.

**Comment by kaso17** [comment]
fixed via f299cf3ac363cb51e13feefb349c96f3aafdbf97
This will break every few weeks until they fix their certificates, please point the staff at the incomplete CA chain issue: https://www.ssllabs.com/ssltest/analyze.html?d=ilcorsaronero.info&latest

**Comment by bubez81** [comment]
ok thank you!

**Comment by Feniz** [comment]
It seems it's broken again.

**Comment by kaso17** [comment]
@Feniz certificate didn't change, thumbprint is still 8D4B5B12D12693F29271C6A0A65ACB3DF6029202

**Comment by Feniz** [comment]
@kaso17 thank you for answering. I have this error while testing the Indexer:
![2017-11-14 16_27_23-jackett](https://user-images.githubusercontent.com/7456876/32788201-dbffbad6-c958-11e7-8a80-fd9ce8584556.png)

Is there something I have to set manually?

**Comment by Feniz** [comment]
I checked and using  0.8.423.0 it works on windows 10 but not on the Synology DS213

**Comment by kaso17** [comment]
@Feniz Can you post a the enhanced log (output after startup and then the failed request(s))

**Comment by Feniz** [comment]
@kaso17 here it is, it tooke me some time to figure out how to retrieve the log. I only replaced the address of my syno with XXX. Thank you!
[log.txt](https://github.com/Jackett/Jackett/files/1475942/log.txt)

**Comment by kaso17** [comment]
In the log you're using the legacy libcurl webclient which doesn't support the certificates pinning feature.
I assume this problem was solved by upgrading to mono 5.8 on your syno?

**Comment by Feniz** [comment]
Unfortunately it wasn't. Actually the log I posted is after I installed mono 5.8.

**Comment by kaso17** [comment]
then please try starting jackett with the "-c httpwebclient" argument

**Comment by Feniz** [comment]
I don't know how to execute start jackett from command line... I can SSH access but I can't manage to find the correct mono path. The command I use is:
`exec /usr/bin/mono --debug /volume1/@appstore/jackett/share/jackett/JackettConsole.exe -c httpwebclient`
Sorry and thanks for the help :-)

**Comment by Feniz** [comment]
Hi, I am not sure I am using the correct syntax:

`exec /volume1/@appstore/mono/bin/mono --debug /volume1/@appstore/jackett/share/jackett/JackettConsole.exe -c httpwebclient`

could you please confirm? Thank you

**Comment by garfield69** [comment]
I believe  you need to replace `httpwebclient` with `httpclient` so try using that.

**Comment by Feniz** [comment]
Thank you @garfield69! Now the indexer works properly. Is there a way to insert the `-c httpclient` argument by default so that I do not have to SSH and manually start jackett with the correct argument?

**Comment by Feniz** [comment]
Ok, it seems I managed to find a way. I post here the solution even if it may be obvious; consider that a lot of Synology guys are no IT experts, after all we bought a Synology to take care of all the mess, isn't it? :-)

Software (for the records):
- DSM 6.1.4-15217 Update 1
- Mono 5.8.0.22-9
- Jackett 0.8.436.0

DSM --> Conrol Panel --> Task Scheduler

Create a New Task (scheduled or triggered as you like, maybe triggered on start is the best), owned by admin (not root as jackett will return an error if run by root, even it will still run):

`exec /volume1/@appstore/mono/bin/mono --debug /volume1/@appstore/jackett/share/jackett/JackettConsole.exe -c httpclient`

save and run it. I hope this can save some time to someone!

**Comment by kaso17** [comment]
@Feniz thank you for testing,
I suspect that BTLS support is not correctly detected.

Can you please try downloading this file to the NAS:
https://www.file-upload.net/download-12824822/MonoBTLSTest.exe.html

And then try to run the following commands and post the output?:
```
/volume1/@appstore/mono/bin/mono .../MonoBTLSTest.exe
MONO_TLS_PROVIDER=btls /volume1/@appstore/mono/bin/mono .../MonoBTLSTest.exe
MONO_TLS_PROVIDER=legacy /volume1/@appstore/mono/bin/mono .../MonoBTLSTest.exe
```

**Comment by Feniz** [comment]
@kaso17 there you are:

`/volume1/@appstore/mono/bin/mono /volume1/@appstore/mono/MonoBTLSTest.exe


Unhandled Exception:
/volume1/@appstore/mono/bin/mono() [0x137070]
/volume1/@appstore/mono/bin/mono() [0x137204]
/volume1/@appstore/mono/bin/mono() [0x13769c]
/volume1/@appstore/mono/bin/mono() [0x138310]
/volume1/@appstore/mono/bin/mono() [0x139078]
/volume1/@appstore/mono/bin/mono() [0x1b7194]

[ERROR] FATAL UNHANDLED EXCEPTION: System.Net.WebException: Error: TrustFailure (The authentication or decryption has failed.) ---> System.IO.IOException: The authentication or decryption has failed. ---> System.IO.IOException: The authentication or decryption has failed. ---> Mono.Security.Protocol.Tls.TlsException: Invalid certificate received from server. Error code: 0xffffffff800b010a
  at Mono.Security.Protocol.Tls.RecordProtocol.EndReceiveRecord (System.IAsyncResult asyncResult) [0x00037] in <ed44b64241d54b8b9efd077a9f94cbd1>:0
  at Mono.Security.Protocol.Tls.SslClientStream.SafeEndReceiveRecord (System.IAsyncResult ar, System.Boolean ignoreEmpty) [0x00000] in <ed44b64241d54b8b9efd077a9f94cbd1>:0
  at Mono.Security.Protocol.Tls.SslClientStream.NegotiateAsyncWorker (System.IAsyncResult result) [0x00071] in <ed44b64241d54b8b9efd077a9f94cbd1>:0
   --- End of inner exception stack trace ---
  at Mono.Security.Protocol.Tls.SslClientStream.EndNegotiateHandshake (System.IAsyncResult result) [0x00032] in <ed44b64241d54b8b9efd077a9f94cbd1>:0
  at Mono.Security.Protocol.Tls.SslStreamBase.AsyncHandshakeCallback (System.IAsyncResult asyncResult) [0x0000c] in <ed44b64241d54b8b9efd077a9f94cbd1>:0
   --- End of inner exception stack trace ---
  at Mono.Security.Protocol.Tls.SslStreamBase.EndRead (System.IAsyncResult asyncResult) [0x0004b] in <ed44b64241d54b8b9efd077a9f94cbd1>:0
  at Mono.Net.Security.Private.LegacySslStream.EndAuthenticateAsClient (System.IAsyncResult asyncResult) [0x0000e] in <4957f27dd94b4398b7d1ac552eecfa25>:0
  at Mono.Net.Security.Private.LegacySslStream.AuthenticateAsClient (System.String targetHost, System.Security.Cryptography.X509Certificates.X509CertificateCollection clientCertificates, System.Security.Authentication.SslProtocols enabledSslProtocols, System.Boolean checkCertificateRevocation) [0x0000e] in <4957f27dd94b4398b7d1ac552eecfa25>:0
  at Mono.Net.Security.MonoTlsStream.CreateStream (System.Byte[] buffer) [0x0007b] in <4957f27dd94b4398b7d1ac552eecfa25>:0
  at System.Net.WebConnection.CreateStream (System.Net.HttpWebRequest request) [0x00073] in <4957f27dd94b4398b7d1ac552eecfa25>:0
   --- End of inner exception stack trace ---
  at System.Net.WebClient.DownloadDataInternal (System.Uri address, System.Net.WebRequest& request) [0x00072] in <4957f27dd94b4398b7d1ac552eecfa25>:0
  at System.Net.WebClient.DownloadString (System.Uri address) [0x00020] in <4957f27dd94b4398b7d1ac552eecfa25>:0
  at System.Net.WebClient.DownloadString (System.String address) [0x00016] in <4957f27dd94b4398b7d1ac552eecfa25>:0
  at MonoBTLSTest.Program.Main (System.String[] args) [0x00006] in <e7383cf752fe438ba871a8fc0fbb0aaf>:0

  *******************************************
  
  MONO_TLS_PROVIDER=btls /volume1/@appstore/mono/bin/mono /volume1/@appstore/mono/MonoBTLSTest.exe

Unhandled Exception:
/volume1/@appstore/mono/bin/mono() [0x137070]
/volume1/@appstore/mono/bin/mono() [0x137204]
/volume1/@appstore/mono/bin/mono() [0x13769c]
/volume1/@appstore/mono/bin/mono() [0x138310]
/volume1/@appstore/mono/bin/mono() [0x139078]
/volume1/@appstore/mono/bin/mono() [0x1b7194]

[ERROR] FATAL UNHANDLED EXCEPTION: System.Net.WebException: Error: ConnectFailure (TLS Support not available.) ---> System.NotSupportedException: TLS Support not available. ---> System.NotSupportedException: No such TLS Provider: `btls'.
  at Mono.Net.Security.MonoTlsProviderFactory.LookupProvider (System.String name, System.Boolean throwOnError) [0x00032] in <4957f27dd94b4398b7d1ac552eecfa25>:0
  at Mono.Net.Security.MonoTlsProviderFactory.CreateDefaultProviderImpl () [0x00019] in <4957f27dd94b4398b7d1ac552eecfa25>:0
  at Mono.Net.Security.MonoTlsProviderFactory.InitializeInternal () [0x0001e] in <4957f27dd94b4398b7d1ac552eecfa25>:0
   --- End of inner exception stack trace ---
  at Mono.Net.Security.MonoTlsProviderFactory.InitializeInternal () [0x00032] in <4957f27dd94b4398b7d1ac552eecfa25>:0
  at Mono.Net.Security.MonoTlsProviderFactory.GetProviderInternal () [0x00010] in <4957f27dd94b4398b7d1ac552eecfa25>:0
  at Mono.Net.Security.MonoTlsStream..ctor (System.Net.HttpWebRequest request, System.Net.Sockets.NetworkStream networkStream) [0x00027] in <4957f27dd94b4398b7d1ac552eecfa25>:0
  at System.Net.WebConnection.CreateStream (System.Net.HttpWebRequest request) [0x00066] in <4957f27dd94b4398b7d1ac552eecfa25>:0
   --- End of inner exception stack trace ---
  at System.Net.WebClient.DownloadDataInternal (System.Uri address, System.Net.WebRequest& request) [0x00072] in <4957f27dd94b4398b7d1ac552eecfa25>:0
  at System.Net.WebClient.DownloadString (System.Uri address) [0x00020] in <4957f27dd94b4398b7d1ac552eecfa25>:0
  at System.Net.WebClient.DownloadString (System.String address) [0x00016] in <4957f27dd94b4398b7d1ac552eecfa25>:0
  at MonoBTLSTest.Program.Main (System.String[] args) [0x00006] in <e7383cf752fe438ba871a8fc0fbb0aaf>:0

  *******************************************
  
  MONO_TLS_PROVIDER=legacy /volume1/@appstore/mono/bin/mono /volume1/@appstore/mono/MonoBTLSTest.exe

Unhandled Exception:
/volume1/@appstore/mono/bin/mono() [0x137070]
/volume1/@appstore/mono/bin/mono() [0x137204]
/volume1/@appstore/mono/bin/mono() [0x13769c]
/volume1/@appstore/mono/bin/mono() [0x138310]
/volume1/@appstore/mono/bin/mono() [0x139078]
/volume1/@appstore/mono/bin/mono() [0x1b7194]

[ERROR] FATAL UNHANDLED EXCEPTION: System.Net.WebException: Error: TrustFailure (The authentication or decryption has failed.) ---> System.IO.IOException: The authentication or decryption has failed. ---> System.IO.IOException: The authentication or decryption has failed. ---> Mono.Security.Protocol.Tls.TlsException: Invalid certificate received from server. Error code: 0xffffffff800b010a
  at Mono.Security.Protocol.Tls.RecordProtocol.EndReceiveRecord (System.IAsyncResult asyncResult) [0x00037] in <ed44b64241d54b8b9efd077a9f94cbd1>:0
  at Mono.Security.Protocol.Tls.SslClientStream.SafeEndReceiveRecord (System.IAsyncResult ar, System.Boolean ignoreEmpty) [0x00000] in <ed44b64241d54b8b9efd077a9f94cbd1>:0
  at Mono.Security.Protocol.Tls.SslClientStream.NegotiateAsyncWorker (System.IAsyncResult result) [0x00071] in <ed44b64241d54b8b9efd077a9f94cbd1>:0
   --- End of inner exception stack trace ---
  at Mono.Security.Protocol.Tls.SslClientStream.EndNegotiateHandshake (System.IAsyncResult result) [0x00032] in <ed44b64241d54b8b9efd077a9f94cbd1>:0
  at Mono.Security.Protocol.Tls.SslStreamBase.AsyncHandshakeCallback (System.IAsyncResult asyncResult) [0x0000c] in <ed44b64241d54b8b9efd077a9f94cbd1>:0
   --- End of inner exception stack trace ---
  at Mono.Security.Protocol.Tls.SslStreamBase.EndRead (System.IAsyncResult asyncResult) [0x0004b] in <ed44b64241d54b8b9efd077a9f94cbd1>:0
  at Mono.Net.Security.Private.LegacySslStream.EndAuthenticateAsClient (System.IAsyncResult asyncResult) [0x0000e] in <4957f27dd94b4398b7d1ac552eecfa25>:0
  at Mono.Net.Security.Private.LegacySslStream.AuthenticateAsClient (System.String targetHost, System.Security.Cryptography.X509Certificates.X509CertificateCollection clientCertificates, System.Security.Authentication.SslProtocols enabledSslProtocols, System.Boolean checkCertificateRevocation) [0x0000e] in <4957f27dd94b4398b7d1ac552eecfa25>:0
  at Mono.Net.Security.MonoTlsStream.CreateStream (System.Byte[] buffer) [0x0007b] in <4957f27dd94b4398b7d1ac552eecfa25>:0
  at System.Net.WebConnection.CreateStream (System.Net.HttpWebRequest request) [0x00073] in <4957f27dd94b4398b7d1ac552eecfa25>:0
   --- End of inner exception stack trace ---
  at System.Net.WebClient.DownloadDataInternal (System.Uri address, System.Net.WebRequest& request) [0x00072] in <4957f27dd94b4398b7d1ac552eecfa25>:0
  at System.Net.WebClient.DownloadString (System.Uri address) [0x00020] in <4957f27dd94b4398b7d1ac552eecfa25>:0
  at System.Net.WebClient.DownloadString (System.String address) [0x00016] in <4957f27dd94b4398b7d1ac552eecfa25>:0
  at MonoBTLSTest.Program.Main (System.String[] args) [0x00006] in <e7383cf752fe438ba871a8fc0fbb0aaf>:0
`

Happy to help testing.

**Comment by kaso17** [comment]
thank you, no BTLS support in that version, some indexers won't work with the mono version.
Please try the new mono packages from the Syno issue https://github.com/SynoCommunity/spksrc/issues/2808

**Comment by Feniz** [comment]
I'm back from a work trip, i see a lot happened on the thread you linked... which one is exactly the mono package version I should test? Thank you

**Comment by kaso17** [comment]
@Feniz please try the new 88f6281 mono 5.4 package from https://github.com/SynoCommunity/spksrc/issues/2808#issuecomment-348180012

**Comment by kaso17** [comment]
And re-run the MonoBTLSTest.exe stuff from above

**Comment by Feniz** [comment]
Solved as reported in #2808. The tracker works as well. Thank you @kaso17 !

Software:
- DSM 6.1.4-15217 Update 2
- Mono 5.4.1.6-9
- Jackett 0.8.469.0
